src/python/dicomifier/dicom_to_nifti/convert.py

```diff
@@ -74,31 +74,32 @@ def get_element(data_set, tag):
                 "-".join(study), "-".join(series), stack_info))
         stacks_converted[series_instance_uid] += 1
 
-        sort(data_sets)
+        sort(keys, data_sets_frame_idx)
+        data_sets = [x[0] for x in data_sets_frame_idx]
+        nifti_img = image.get_image(data_sets_frame_idx, dtype)
+        nifti_meta_data = meta_data.get_meta_data(data_sets_frame_idx)
 
-        nifti_image = image.get_image(data_sets, dtype)
-        nifti_meta_data = meta_data.get_meta_data(data_sets)
-
-        nifti_data.append((nifti_image, nifti_meta_data))
+        nifti_data.append((nifti_img, nifti_meta_data))
 
     # Try to preserve the original stacks order
     nifti_data.sort(key=lambda x: x[1].get("InstanceNumber", [None])[0])
 
-    # Try to merge stacks in each series
-    merged_stacks = []
     series = {}
-    for nitfi_image, nifti_meta_data in nifti_data:
+    for nitfi_img, nifti_meta_data in nifti_data:
         series.setdefault(nifti_meta_data["SeriesInstanceUID"][0], []).append(
-            (nitfi_image, nifti_meta_data))
+            (nitfi_img, nifti_meta_data))
+
+    merged_stacks = []
     for stacks in series.values():
         mergeable = {}
-        for nitfi_image, nifti_meta_data in stacks:
-            geometry = nifti_image.shape + \
-                tuple(nitfi_image.qform.ravel().tolist())
-            mergeable.setdefault(geometry, []).append(
-                (nitfi_image, nifti_meta_data))
-
-        for stack in mergeable.values():
+        for nifti_img, nifti_meta_data in stacks:
```

Comment:
Something weird is happening during the merge: elements with per-frame values (e.g. `ContentTime`, `WindowWidth`) retain a single value in the JSON.